parser-wrapper/interface-impl-amf/src/main/java/org/mule/amf/impl/ParserWrapperAmf.java

```diff
@@ -241,11 +243,15 @@ private String dumpRaml() {
   public void updateBaseUri(IRaml api, String baseUri) {}
 
   public String getAmfModel() {
-    try {
-      return new AmfGraphRenderer().generateString(document).get();
-    } catch (InterruptedException | ExecutionException e) {
-      return e.getMessage();
+    if (amfModel == null) {
+      try {
+        new AmfGraphResolver().resolve(document);
```

Comment:
is this needed? I mean, the result is not assigned to any variable

Reply:
Yes, that's the key of the fix. The document needs to be resolved before rendering it